pymc/model.py

```diff
@@ -1766,7 +1766,7 @@ def set_data(new_data, model=None):
 
     for variable_name, new_value in new_data.items():
         model.set_data(variable_name, new_value)
-
+    return model.set_data(new_data, coords)
```

Comment:
Instead of this, you'll need to edit the previous line.

Also make sure to add a test

Reply:
Ok i will do this changes at the earliest